File: src/test/java/pages/LoginPage.java
package pages;

import com.microsoft.playwright.Page;

public class LoginPage extends BaseTest{
    private Page page;

    private String username = "//input[@name='username']";
    private String password = "//input[@name='password']";
    private String loginBtn = "//input[@value='Log In']";

    public LoginPage(Page page) {
        this.page = page;
    }

    public void navigate() {
        page.navigate("https://parabank.parasoft.com/parabank/index.htm");
    }

    public void login(String user, String pass) {
        page.fill(username, user);
        page.fill(password, pass);
        page.click(loginBtn);
    }
}

File: src/pages/login.page.ts
import { Page } from 'playwright';

export class LoginPage {
  private readonly username = "//input[@name='username']";
  private readonly password = "//input[@name='password']";
  private readonly loginBtn = "//input[@value='Log In']";

  constructor(private readonly page: Page) {}

  async navigate(): Promise<void> {
    await this.page.goto('https://parabank.parasoft.com/parabank/index.htm');
  }

  async login(user: string, pass: string): Promise<void> {
    await this.page.fill(this.username, user);
    await this.page.fill(this.password, pass);
    await this.page.click(this.loginBtn);
  }
}
